test/unit/ECDSAStakeRegistryUnit.t.sol

```diff
@@ -827,13 +834,239 @@ contract ECDSAStakeRegistryTest is ECDSAStakeRegistrySetup {
+
+        // Register first operator with the signing key
+        vm.prank(operator1Address);
+        newRegistry.registerOperatorWithSignature(operatorSignature, sharedSigningKey);
+
+        // Try to register second operator with the same signing key
+        // This should now revert with SigningKeyAlreadyInUse
+        vm.prank(operator2Address);
+        vm.expectRevert(abi.encodeWithSignature("SigningKeyAlreadyInUse()"));
+        newRegistry.registerOperatorWithSignature(operatorSignature, sharedSigningKey);
+    }
+    
+    function test_ReuseSigningKeyAfterUpdate() public {
+        // Create a new registry instance for this test to avoid conflicts with the setup
+        IStrategy mockStrategy = IStrategy(address(0x1234));
+        IECDSAStakeRegistryTypes.Quorum memory quorum = IECDSAStakeRegistryTypes.Quorum({
+            strategies: new IECDSAStakeRegistryTypes.StrategyParams[](1)
+        });
+        quorum.strategies[0] = 
+            IECDSAStakeRegistryTypes.StrategyParams({strategy: mockStrategy, multiplier: 10000});
+        ECDSAStakeRegistry newRegistry = new ECDSAStakeRegistry(IDelegationManager(address(mockDelegationManager)));
+        newRegistry.initialize(address(mockServiceManager), 100, quorum);
+        
+        // Create two different operators
+        address operator1Address = address(0xABCD);
+        address operator2Address = address(0xDCBA);
+        
+        // Two different signing keys
+        address signingKey1 = address(0x1111);
+        address signingKey2 = address(0x2222);
+
+        ISignatureUtilsMixinTypes.SignatureWithSaltAndExpiry memory operatorSignature;
+
+        // Step 1: Register operator1 with signingKey1
+        vm.prank(operator1Address);
+        newRegistry.registerOperatorWithSignature(operatorSignature, signingKey1);
+        
+        // Verify operator1 is associated with signingKey1
+        assertEq(
+            newRegistry.getLatestOperatorForSigningKey(signingKey1),
+            operator1Address,
+            "operator1 should be associated with signingKey1"
+        );
+        
+        // Step 2: Update operator1's signing key to signingKey2
+        vm.prank(operator1Address);
+        newRegistry.updateOperatorSigningKey(signingKey2);
+        
+        // Verify operator1 is now associated with signingKey2
+        assertEq(
+            newRegistry.getLatestOperatorForSigningKey(signingKey2),
+            operator1Address,
+            "operator1 should be associated with signingKey2 after update"
+        );
+        
+        // Verify signingKey1 is no longer associated with any operator
+        assertEq(
+            newRegistry.getLatestOperatorForSigningKey(signingKey1),
+            address(0),
+            "signingKey1 should not be associated with any operator after update"
+        );
+        
+        // Step 3: Register operator2 with signingKey1 (which should be allowed now)
```

Comment:
Okay, so this is the safe "reuse signing key without conflict" path?
Not sure I think this is good behavior, but at least it doesn't overwrite any data.